torch/_inductor/ir.py

```diff
@@ -3862,6 +3869,9 @@ def __init__(self, *, kernel_idx, grid, kernel_args):
         self.kernel_idx = kernel_idx
         self.grid = grid
 
+        kernel, _ = self.get_kernel_and_configs()
+        self.ordered_kwargs_for_cpp_kernel = kernel.arg_names
```

Comment:
It seems to be pretty hacky to rely on ```ordered_kwargs_for_cpp_kernel``` to generate kernel args indirectly. Wondering if we could implement a ```codegen_args``` method for ```UserDefinedTritonKernel```, where we return arg strings.

Reply:
I'm not following. It is not possible to return all strings because then sympy expr can no longer be distinguished. Look at https://github.com/pytorch/pytorch/blob/562c4ae4bcda0cdede34f6cf9f35b6feb61e4268/torch/_inductor/codegen/wrapper.py#L2468-L2501